pkg/codegen/tmpl/core_resource.tmpl

```diff
@@ -1,6 +1,29 @@
 package {{ .PackageName }}
 
-// Resource is the wire representation of {{ .KindName }}. (TODO be better)
+import (
+	"github.com/grafana/grafana/pkg/kinds"
+)
+
+// Resource is the kubernetes style representation of {{ .KindName }}. (TODO be better)
+type K8sResource = kinds.GrafanaResource[Spec, Status]
+
+// NewResource creates a new instance of the resource with a given name (UID)
+func NewK8sResource(name string, s *Spec) K8sResource {
+	return K8sResource{
+		Kind:       "{{ .KindName }}",
+		APIVersion: "alphav1",
```

Comment:
^^ hardcoded version!!!  again, that should be sorted out somehow

Reply:
It's right at hand in the jenny - `sfg.Schema.Version()`. Assign it to a new prop named `Version` in `tvar_resource`, then this change will do the trick:
```suggestion
		APIVersion: "v{{ .Version }}",
```